flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -237,66 +253,11 @@ void testDelayedGrantCallAfterContenderBeingDeregisteredAgain() throws Exception
         };
     }
 
-    /**
-     * Test to cover the issue described in FLINK-31814. This test could be removed after
-     * FLINK-31814 is resolved.
-     */
-    @Test
-    void testOnRevokeCallWhileClosingService() throws Exception {
-        final TestingLeaderElectionDriver.TestingLeaderElectionDriverFactory driverFactory =
-                new TestingLeaderElectionDriver.TestingLeaderElectionDriverFactory(
-                        LeaderElectionEventHandler::onRevokeLeadership);
-
-        try (final DefaultLeaderElectionService testInstance =
-                new DefaultLeaderElectionService(driverFactory)) {
-            testInstance.startLeaderElectionBackend();
-
-            final TestingLeaderElectionDriver driver = driverFactory.getCurrentLeaderDriver();
-            assertThat(driver).isNotNull();
-
-            driver.isLeader();
-
-            final LeaderElection leaderElection =
-                    testInstance.createLeaderElection(createRandomContenderID());
-            final TestingContender contender =
-                    new TestingContender("unused-address", leaderElection);
-            contender.startLeaderElection();
-
-            contender.waitForLeader();
-
-            leaderElection.close();
-
-            contender.throwErrorIfPresent();
-        }
-    }
-
-    @Test
-    void testStopWhileHavingLeadership() throws Exception {
```

Comment:
Why is this test being removed?

Reply:
I tried to put test removals into their own commits to be able to give the reasoning in the commit message. For this specific test it is:
```
[test] Removes testStopWhileHavingLeadership: This test case is covered by 
- testDelayedGrantCallAfterContenderRegistration which checks delayed grant event processing (the first part of the test granting leadership)
- testOnRevokeLeadershipIsTriggeredAfterLeaderElectionBeingStop which checks that the leadership is revoked in the contender after the close is called (the second part of the test revoking the leadership in the contender)
```